src/context/Search/suggestions/match/getMatch.ts

```diff
@@ -0,0 +1,40 @@
+import { CurriculumData } from "../oakCurriculumData";
+
+import { rankMatches } from "./rankMatches";
+
+export type MatchResult = { slug: string; matched: string };
+
+export const getMatch = (
+  query: string,
+  data: CurriculumData[],
+): MatchResult | undefined => {
+  const matches: Array<MatchResult> = [];
+
+  data.forEach((datum) => {
+    const slugRegex = new RegExp(datum.slug, "i");
+    const matchesSlug = query.match(slugRegex);
```

Comment:
In case you hadn't considered it, another way to do this is `slugRegex.toLowerCase === datum.slug.toLowerCase`

I don't think that either approach is better or worse so long as we trust the input data. Building regexes from user data could potentially allow a ReDoS which I know SonarCloud is keen to flag up!

Reply:
I've just realised that you're matching for substrings so in that case it would obviously be `query.toLowerCase().includes(datum.slug.toLowerCase())`